src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -131,10 +131,69 @@ public PackageArchiveReader BuildPackage(PackageBuilder builder, string outputPa
                 PrintVerbose(outputPath, builder);
             }
 
+            if (_packArgs.OutputResolvedNuSpec)
+            {
+                WriteResolvedNuSpecToPackageOutputDirectory(outputPath, builder);
+            }
+
+            if (_packArgs.OutputSHA512PackageHash)
+            {
+                WriteSHA512PackageHash(outputPath, builder);
+            }
+
             _packArgs.Logger.LogMinimal(String.Format(CultureInfo.CurrentCulture, Strings.Log_PackageCommandSuccess, outputPath));
             return new PackageArchiveReader(outputPath);
         }
 
+        /// <summary>
+        /// Writes the resolved NuSpec file to the package output directory.
+        /// </summary>
+        /// <param name="packageOutputPath">The output package path</param>
+        /// <param name="packageBuilder">The package builder</param>
+        private void WriteResolvedNuSpecToPackageOutputDirectory(string packageOutputPath, PackageBuilder packageBuilder)
+        {
+            string packageOutputDirectory = Path.GetDirectoryName(packageOutputPath);
+            string resolvedNuSpecOutputPath = Path.Combine(packageOutputDirectory, packageBuilder.Id + PackagingConstants.ManifestExtension);
+
+            if (string.Equals(_packArgs.Path, resolvedNuSpecOutputPath, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new PackagingException(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        "Unable to output resolved nuspec file because it would overwrite the original at '{0}'",
+                        _packArgs.Path));
+            }
+
+            using (FileStream fileStream = new FileStream(resolvedNuSpecOutputPath, FileMode.Create))
+            {
+                using (StreamWriter streamWriter = new StreamWriter(fileStream, System.Text.Encoding.UTF8))
+                {
+                    packageBuilder.PackageManifest.ResolvedManifest.Save(streamWriter, System.Xml.Linq.SaveOptions.None);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes the sha512 package hash file to the package output directory
+        /// </summary>
+        /// <param name="packageOutputPath">The output package path</param>
+        /// <param name="packageBuilder">The package builder</param>
+        private void WriteSHA512PackageHash(string packageOutputPath, PackageBuilder packageBuilder)
+        {
+            string packageOutputDirectory = Path.GetDirectoryName(packageOutputPath);
```

Comment:
@emgarten there must already be a helper function that does this , right? in install/restore? can you point nick to it?